### PR TITLE
feat: add a new plugin newyork_times

### PR DESCRIPTION
**Pull request by LinkW77**
# Pull Request

## PR Description

Add a new plugin New York Times with Search API and Top Stories API

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Performance enhancement
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other, please describe:

## Checklist

Before submitting this PR, please make sure:

- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md) guidelines.
- [x] I have tested my changes locally to ensure they are effective.
- [x] I have updated the necessary documentation (if applicable).

### CODE REVIEW

File: plugin/bundles/nytimes/plugins/news_search/plugin.py
Comment (DynamesC) [inline plugin/bundles/nytimes/plugins/news_search/plugin.py:1]: Add input format validation. If begin_date or end_date does not meet with the required format, raise request validation error.

File: plugin/bundles/nytimes/resources/i18n/en.yml
Comment (DynamesC) [inline plugin/bundles/nytimes/resources/i18n/en.yml:1]: Update description for start and end date for better clarity.

My personal opinion:

Start Date: The earliest publication date for articles returned by the search. Articles published before this date are not included in the results.
End Date: The latest publication date for articles returned by the search. Articles published after this date are not included in the results.

**Review by DynamesC — CHANGES_REQUESTED** [review]

**Review by DynamesC — APPROVED** [review]
LGTM

**Review by taskingaijc — APPROVED** [review]
LGTM

**Pull request merged**